README.md

```diff
@@ -52,7 +52,58 @@ To see additional options, visit:
     
 ### Kafka
 
-*TODO*
+ - In the *dev* mode, the application expects a Kafka broker running at `localhost:9092`.
+ - In the *prod* mode, you have to provide an environment variable KAFKA_BOOTSTRAP_SERVERS pointing to Kafka brokers
+
+Kafka storage implementation uses the following Kafka API / architecture
+
+ - Storage producer to forward REST API HTTP requests to Kafka broker
+ - Storage consumer to handle previously sent  REST API HTTP requests as Kafka messages
+ - Snapshot producer to send current state's snapshot to Kafka broker
+ - Snapshot consumer for initial (at application start) snapshot handling
+
+We already have sensible defaults for all these things, but they can still be overridden or added by adding appropriate properties to app's configuration. The following property name prefix must be used:
+
+ - Storage producer: registry.kafka.storage-producer.
+ - Storage consumer: registry.kafka.storage-consumer.
+ - Snapshot producer: registry.kafka.snapshot-producer.
+ - Snapshot consumer: registry.kafka.snapshot-consumer.
+
+We then strip away the prefix and use the rest of the property name in instance's Properties.
+
+e.g. registry.kafka.storage-producer.enable.idempotence=true --> enable.idempotence=true
+
+For the actual configuration options check (although best config docs are in the code itself):
+ - [Kafka configuration](https://kafka.apache.org/documentation/)
+
+### Streams
+
+Streams storage implementation goes beyond plain Kafka usage and uses Kafka Streams to handle storage in a distributed and fault-tolerant way.
+
+ - In the *dev* mode, the application expects a Kafka broker running at `localhost:9092`.
+ - In the *prod* mode, you have to provide an environment variable KAFKA_BOOTSTRAP_SERVERS pointing to Kafka brokers and APPLICATION_ID to name your Kafka Streams application
+
+Both modes require 2 topics: storage topic and globalId topic. This is configurable, by default we use storage-topic and global-id-topic names.
+
+Streams storage implementation uses the following Kafka (Streams) API / architecture
+
+ - Storage producer to forward REST API HTTP requests to Kafka broker
+ - Streams input KStream to handle previously sent  REST API HTTP requests as Kafka messages
+ - Streams KStream to handle input's KStream result
+ - Both KStreams use KeyValueStores to keep current storage state
```

Comment:
I think it would be helpful to mention the kinds of messages you're storing in the KStreams so that the reader can more easily understand how the data is stored and mapped.

Reply:
OK, will add some info about this.

Reply:
I've added some basic and simple info on how the data is structured and how we handle it.

Reply:
That's perfect, thanks.